context/context.go

```diff
@@ -55,7 +55,13 @@ func GetConfig(ctx context.Context) *conf.Configuration {
 
 // WithCoupons adds the coupon cache to the context based on the site URL.
 func WithCoupons(ctx context.Context, config *conf.Configuration) context.Context {
-	return context.WithValue(ctx, couponsKey, coupons.NewCouponCacheFromURL(config))
+	cache, err := coupons.NewCouponCacheFromURL(config)
+	if err != nil {
+		// We don't have access to the logger here, but it's better to say something than nothing
```

Comment:
You should just be able to return the error. Or you could pass in the request to the signatures above this and pull the logger? 

https://github.com/netlify/gocommerce/blob/dfaf9f462663096f468cbd1c8025e8226a6068e9/api/log.go#L60-L66